crates/wasmtime/src/engine.rs

```diff
@@ -783,6 +783,29 @@ impl Engine {
         )
     }
 
+    /// Loads a `CodeMemory` from the specified memory region without copying
+    ///
+    /// The `expected` marker here is whether the bytes are expected to be
+    /// a precompiled module or a component
+    ///
+    /// # Safety
+    ///
+    /// As there is no copy here, the runtime will be making direct use
+    /// of the provided memory and may read or write to this memory.  As such,
+    /// any outside writes to this memory region will result in undefined and
+    /// likely very undesirable behavior.
```

Comment:
Maybe also mention that the pointer must be valid for `size` bytes and must point to the contents of a `.cwasm` file or result of `{Module,Component}::serialize`-style APIs.